### ISSUE TITLE
Initializing sf::SoundBuffer crashes whole program in llvm clang UCRT

### ISSUE THREAD

**Issue by PrashanthKumar0** [SFML/SFML#3145]
### Prerequisite Checklist

- [X] I searched for [existing issues](https://github.com/search?q=repo%3ASFML%2FSFML&type=issues) to prevent duplicates
- [X] I searched for [existing discussions on the forum](https://www.google.com/search?q=site%3Ahttps%3A%2F%2Fen.sfml-dev.org) to prevent duplicates
- [X] I am here to report an issue and not to just ask a question or look for help (use the [forum](https://en.sfml-dev.org/forums/index.php#c3) or [Discord](https://discord.gg/nr4X7Fh) instead)

### Describe your issue here

There's something wrong with `sf::SoundBuffer` initializing it causes whole thing crash  without any clue.
I tried debugging setting breakpoint and it seems like I couldn't
see below for a better understanding of what's going on...

Here's a bare minimum code.

```cpp
#include <SFML/Audio.hpp>
#include <iostream>

int main()
{
    std::cout << "SB test\n" << std::flush;    
    sf::SoundBuffer sb{};
    std::cout << "PASS" << std::flush;    
}
```

I have cloned branch `2.6.x`

Here's my compiler Details and demo .
![image](https://github.com/SFML/SFML/assets/31966594/2016798c-fb0d-463e-800c-f6a4c7d03d26)

### Your Environment

- OS / distro / window manager:  Windows 10
- SFML version: 2.6.x (git cloned)
- Compiler / toolchain:  clang version 18.1.0rc ( https://github.com/llvm/llvm-project.git 6c90f8dd546334b01c9a86387950e8c72d459f1e )
- Special compiler / CMake flags: Nothing

### Steps to reproduce

```cpp
#include <SFML/Audio.hpp>
#include <iostream>

int main()
{
    std::cout << "SB test\n" << std::flush;    
    sf::SoundBuffer sb{};
    std::cout << "PASS" << std::flush;    
}
```

### Expected behavior

Should atleast print "SB test" but it doesn't.

### Actual behavior

Atleast it should run and print something on screen.

**Comment by Bambo-Borris** [comment]
Isn't UCRT known to cause issues with SFML since the dependencies are compiled with MSVCRT

**Comment by eXpl0it3r** [comment]
We fixed that with #3115 for 2.6.x, backmerge still pending.

Did you build it in debug mode?

**Comment by eXpl0it3r** [comment]
This has been back-merged by now.
